src/WordPress/Blueprints/Runner/Step/RmStepRunner.php

```diff
@@ -2,18 +2,27 @@
 
 namespace WordPress\Blueprints\Runner\Step;
 
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+use WordPress\Blueprints\BlueprintException;
 use WordPress\Blueprints\Model\DataClass\RmStep;
 
 
-class RmStepRunner extends BaseStepRunner {
-	/**
-	 * @param RmStep $input
-	 */
-	function run( RmStep $input ) {
-		// @TODO: Treat $input->path as relative path to the document root (unless it's absolute)
-		$success = unlink( $input->path );
-		if ( ! $success ) {
-			throw new \Exception( "Failed to remove the file at {$input->path}" );
-		}
-	}
-}
+class RmStepRunner extends BaseStepRunner
+{
+    /**
+     * @param RmStep $input
+     */
+    function run(RmStep $input) {
+        $resolvedPath = $this->getRuntime()->resolvePath($input->path);
+        $fileSystem = new Filesystem();
+        if (false === $fileSystem->exists($resolvedPath)) {
+            throw new BlueprintException("Failed to remove $resolvedPath: the directory or file does not exist.");
+        }
+        try {
+            $fileSystem->remove($resolvedPath);
+        } catch (IOException $exception) {
+            throw new BlueprintException("Failed to remove the directory or file at $resolvedPath", 0, $exception);
+        }
+    }
+}
```

Comment:
Nitpick: there's no empty line at the end of the file which GitHub marks like this:

![CleanShot 2024-02-21 at 14 42 35@2x](https://github.com/WordPress/blueprints/assets/205419/267fd3ab-00d0-4093-bb89-7ed8a76de530)

Reply:
Done.